Assets/Scripts/Gameplay/ConnectionManagement/ConnectionState/ClientReconnectingState.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.Collections;
+using System.Threading.Tasks;
+using Unity.Multiplayer.Samples.BossRoom.Shared;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Net.UnityServices.Lobbies;
+using Unity.Multiplayer.Samples.Utilities;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    /// <summary>
+    /// Connection state corresponding to a client attempting to reconnect to a server. It will try to reconnect a
+    /// number of times defined by k_NbReconnectAttempts. If it succeeds, it will transition to the Connected state. If
+    /// not, it will transition to the Offline state.
+    /// </summary>
+    public class ClientReconnectingState : OfflineState
```

Comment:
(I'm not saying we should do this) any reason why connecting is an offline state but reconnecting isn't?

Reply:
Reconnecting inherits from OfflineState because it handles some connection methods that are present in the OfflineState.

Reply:
yep, that's not a valid reason for having it inherit. Think of inheritance as "class A IS a classB". If we need common utils or things like that, let's extract those in a common util. It needs to make sense to say that class A IS a classB, else you end up with a class that contains behaviour it shouldn't.
To me, from a design perspective, it'd make sense to have both connecting and reconnecting "IS" an offline state (as they are not online yet). They could also be their own thing. But we should be purposeful about it not do it "just because it has some common methods".